implementation/common/pom.xml

```diff
@@ -57,6 +57,10 @@
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
     </dependency>
+    <dependency>
```

Comment:
We don't need another dependency IMHO

Reply:
If we want to be able to do property-based tests, then we need. Given it's just a test dependency, it would be shame to resign from property-based tests. But it's your call.

Reply:
If we want to be able to do property-based tests, then we do need it. Given it's just a test dependency, it would be shame to resign from this. But it's your call.

Reply:
@0rzech I see, I missed you meant to use it for tests only. These tests look fine, but for some reasons this dependency requires a dedicated exclusion in .gitignore. I appreciate you like the style offered but me being quite conservative, I'd say we can easily enough, even if it will be more verbose, to write similar tests without it.

Let me ask Mike. @MikeEdgar, do you think we should bring it or just use existing test dependencies ?

Reply:
`.jqwik-database` is a file to store data of previous runs. Given how property-based testing works with shrinking etc., I don't think we should try to emulate it ourselves.

Reply:
At the same time, we don't really need to introduce it IMHO if it keeps some local state after runs

Reply:
It's helpful if you want to re-run failing properties, but I can disable it.

Reply:
The whole point of property-based testing is not to rely on one or few specific input examples, but to have tests that cover large space of inputs and help with pinpointing specific errors. `@ParameterizedTest` was not made for this.

IMHO, it's worth investing even more into property-based testing where appropriate, especially in security-related project. For instance, property-based tests could be used to randomize claim headers in JWTs, instead of just making sure that headers work fine for `https://issuer.com` issuer and `customHeader` with `custom-header-value`, because this is what current tests actually prove.

The reason I was hit with the bug was that the tests did not cover the code path my project triggered, that's why I've added tests covering all other signature algorithms. To be fair, though, because all possible algorithms is a relatively small finite input space, `@ParameterizedTest` could do it. It's opposite for invalid inputs in this particular case. 

If you are worried about `jqwik` being some random library, then here's [Oracle's Java Magazine](https://blogs.oracle.com/javamagazine/post/know-for-sure-with-property-based-testing) article about it from almost 5 years ago. Jqwik has also plugins for Spring Boot and Micronaut, and even someone wanted to add jqwik extension to Quarkus, though - unfortunately - to no avail it seems ([here](https://github.com/quarkusio/quarkus/discussions/29834) and [here](https://github.com/quarkusio/quarkus/discussions/29916)), and [here's](https://github.com/apache/kafka/blob/55d38efcc5505a5a1bddb08ba05f4d923f8050f9/raft/src/test/java/org/apache/kafka/raft/internals/RecordsIteratorTest.java) Apache Kafka using `@ParameterizedTest`, `@Property` and normal `@Test` in a single test class no problem.

Having said all that, I fully respect it's your project, not mine. I just hope that at least most arguments are taken into consideration, that's it.

May I ask to help me with understanding why it is such an issue with adding this library, though? I genuinely don't get it, given it's not code that will run in production anyway and the jqwik project seems to have been started over 8 years ago. I may lack some context, though. It's been years since I was programming Java for a living and I'm completely new to Quarkus.

Reply:
I think for me it's less about a new test dependency and more about having a small subset of the tests using a different paradigm. Based on what I've seen/read, I'm not really opposed to something like property-based testing per se, but rather it might be confusing to do things multiple ways within a single project.

Another possible "gotcha" is IDE support for jqwik. It might be fine, but just guessing it might trip up Eclipse's JUnit runner (for example).

Reply:
Thanks for your reply!

You can't use property-based testing only. Not everything makes sense that way, because not every test input is large.

The same way one wouldn't want to do everything with builder pattern, because most constructors have short list of parameters. And the opposite - one wouldn't want to resign from builder pattern completely just to not confuse developers with doing things multiple ways in a single project. It's a matter of familiarity.

I don't know how Eclipse will react, but jqwik is an alternative engine for Junit 5 platform, so there shouldn't be problems with running tests and reporting. Perhaps one would have to make Eclipse aware of jqwik's test annotations, so that the code won't be marked as unused. I don't know, I tend not to write code to make IDE's happy, but to adjust IDEs to the code. ;) Nevertheless, IntelliJ was able to run jqwik tests pretty fine.